src/domain/gpu/Payloads.ts

```diff
@@ -0,0 +1,9 @@
+export interface DataInMessagePayload {
+  pattern: Uint8Array;
+  text: Uint8Array;
+}
+
+export interface DataOutMessagePayload {
+  matches: Array<number>;
+  error: Error;
+}
```

Comment:
I will refactor this after merge alongside with proper scheduler with shared interface.